File: 0066-plus-one/0066-plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    digits.reverse();
    let carry = 1 ;
    let i = 0 ;
    while(i < digits.length){
        let sum = digits[i] + carry ;
        carry = Math.floor(sum/10) ;
        digits[i] = sum%10 ;
        if(carry == 0)break ;
        i++ ;
    }
    
    if(carry !== 0)digits.push(carry) ;
    return digits.reverse();
};

File: 0066-plus-one/0066-plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1 ;
        for(int i = digits.length - 1 ; i >= 0 ; i--){
            int s = digits[i]+carry ;
            digits[i] = s%10 ;
            carry = s/10 ;
        }
        
        if(carry == 0)return digits ;
        
        int[] newArr = new int[digits.length+1];
        newArr[0] = carry ;
        System.arraycopy(digits, 0, newArr, 1, digits.length);
        
        return newArr ;
    }
}